comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -53,11 +53,14 @@ type collector struct {
 	// taskCollectionEnabled is a flag to enable detailed task collection
 	// if the flag is enabled, the collector will query the latest metadata endpoint, currently v4, for each task
 	// that is returned from the v1/tasks endpoint
-	taskCollectionEnabled bool
-	taskCollectionParser  util.TaskParser
-	taskCache             *cache.Cache
-	taskRateRPS           int
-	taskRateBurst         int
+	taskCollectionEnabled        bool
+	taskCollectionParser         util.TaskParser
+	taskCache                    *cache.Cache
+	taskRateRPS                  int
+	taskRateBurst                int
+	metadataRetryInitialInterval time.Duration
+	metadataRetryMaxElapsedTime  time.Duration
+	metadataRetryTimeoutFactor   int
```

Comment:
Following some comments below, would it be possible to directly store the client here instead of only the parameters used to create it ?

```suggestion
	client                       v3or4.Client
```